src/components/Syntaxes.jsx

```diff
@@ -24,30 +25,51 @@ class Syntaxes extends React.Component {
     this.props.onClick(syntax);
   }
 
-  render() {
+  onSyntaxSearch(e) {
+    this.setState({ searchSyntax: e.target.value.trim() });
+  }
+
+  getSyntaxes() {
+    let { syntaxes } = this.props;
+
+    if (this.state.searchSyntax) {
+      syntaxes = this.filterSyntaxes(regExpEscape(this.state.searchSyntax));
+    }
+
+    return syntaxes.map((syntax, index) => {
```

Comment:
Hm.. frankly speaking, I'd prefer to keep this `map()` call and *component instantiation* where it was before - in `render()`. Why? Because it's somehow easier to navigate knowing that everything that affects DOM is in `render()`. So if you don't mind, can you please move this back to render and keep `getSyntaxes()` as a helper that only filters syntaxes (i.e. so getSyntaxes is merged with filterSyntaxes)?

Reply:
Well, actually, if you both (@AinuraAliieva and @olha-kurkaiedova) think that my suggestion is crap and nonsense - let's go with it. :)

Reply:
In that case `.render()` can be immense in size.
And in my opinion it is better to separate the logic into small pieces (will make the code more readable)